### PR TITLE
Add opacity tests & Fixes opacity

### PR DESCRIPTION
**Pull request by PsiOmegaDelta**
`RecalculateOpacity()` should be macroable.

### CODE REVIEW

**Comment by Lohikar** [comment]
I don't think there's any value in making it a macro though, and it's not a trivial proc.

File: code/unit_tests/observation_tests.dm [hunk 1]
```diff
@@ -348,3 +349,104 @@
 
 	qdel(event_source)
 	return 1
+
+/datum/unit_test/observation/opaque_counter_checks
+	name = "Observation: Opacity - Validate opaque_counter"
+
+/datum/unit_test/observation/opaque_counter_checks/start_test()
+	var/list/landmark_types = list(/obj/effect/landmark/proximity_spawner = 0, /obj/effect/landmark/proximity_wall = 1)
+	var/list/object_types = list(null, /obj/unit_test/transparent = 0, /obj/unit_test/opaque = 1)
```
Comment (PsiOmegaDelta) [inline code/unit_tests/observation_tests.dm]: Prettified formatting locally

File: code/unit_tests/proximity_tests.dm [hunk 2]
```diff
@@ -36,6 +55,48 @@
...
+	wall.set_opacity(FALSE)
+
+/datum/unit_test/proximity/visibility/shall_see_the_number_of_expected_turfs_after_opacity_changes
+	name = "Shall see the number of expected turfs after opacity changes"
+	expected_number_of_turfs_by_trigger_type = list(
+		/datum/proximity_trigger/line = 5,
+		/datum/proximity_trigger/square = 82
+	)
+
+/datum/unit_test/proximity/visibility/shall_see_the_number_of_expected_turfs_after_opacity_changes/BeforeSetTrigger()
```
Comment (PsiOmegaDelta) [inline code/unit_tests/proximity_tests.dm]: Looks like I copy-pastes badly around here. This is why you don't copy-paste.

**Comment by afterthought2** [comment]
Putting the do not merge tag to avoid confusion; you can take it off when it's ready to go.

**Comment by PsiOmegaDelta** [comment]
Tentatively hopeful that the full opacity recalculation changes have fixed any lingering issues proximity listeners not always reacting as they should. During testing turfs had all kinds of unexpected opaque counter values, this has now been resolved.

If not, additional tests could include creating and moving opaque objects and validating expectations then.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**